devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -239,6 +233,30 @@ public void execute(Task test) {
         });
     }
 
+    private void registerConditionalDependencies(Project project) {
+        ConditionalDependenciesEnabler conditionalDependenciesEnabler = new ConditionalDependenciesEnabler(project);
+        ApplicationDeploymentClasspathBuilder deploymentClasspathBuilder = new ApplicationDeploymentClasspathBuilder(
+                project);
+        project.getConfigurations().getByName(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME).getIncoming()
+                .beforeResolve((dependencies) -> {
+
+                    Set<ExtensionDependency> implementationExtensions = conditionalDependenciesEnabler
+                            .declareConditionalDependencies(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME);
+                    deploymentClasspathBuilder.createBuildClasspath(implementationExtensions,
+                            JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME, true);
+                    Set<ExtensionDependency> devModeExtensions = conditionalDependenciesEnabler
+                            .declareConditionalDependencies(DEV_MODE_CONFIGURATION_NAME);
+                    deploymentClasspathBuilder.createBuildClasspath(devModeExtensions, DEV_MODE_CONFIGURATION_NAME, false);
```

Comment:
Just to clarify, are we going to be resolving the dev classpath even if the launch mode isn't DEV?

Reply:
This configuration only contains specific `quarkusDev` dependency. But even for test we are going to resolve them. I don't think we have a way to get the launch mode here.

Reply:
We create DEV_MODE_CONFIGURATION_NAME, could we handle it in the same manner as the COMPILE_CLASSPATH_CONFIGURATION_NAME and TEST_COMPILE_CLASSPATH_CONFIGURATION_NAME? i.e. do the beforeResolve thing?

Reply:
That’s true, we can definitely do that in the before resolve. I will test that

Reply:
fixed !